### PR TITLE
Bug 1859004: Sometimes the eventrouter couldn't gather event logs: switch to go mod and update k8s apimatcher version

### PR DESCRIPTION
**Pull request by vparfonov**
Switch to the `go modules`.
Update k8s `apimatcher` version for fixing network issue with forcibly closed connection. 
More details here: https://github.com/kubernetes/kubernetes/pull/73277  
Signed-off-by: Vitalii Parfonov <vparfono@redhat.com>

### CODE REVIEW

**Review by jcantrill — COMMENTED** [review]
/approve

File: .idea/.gitignore [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Default ignored files
```
[the remark this answers is not in the document]
Comment (vparfonov) [reply .idea/.gitignore]: done

**Comment by jcantrill** [comment]
Manually merging as dependents test pass https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_origin-aggregated-logging/2080/pull-ci-openshift-origin-aggregated-logging-master-cluster-logging-operator-e2e/1371539619488731136

**Review by jcantrill — COMMENTED** [review]
/lgtm

**Pull request merged**